### PR TITLE
LibLine: Correctly handle line content overflow when on last line

### LINKED ISSUE #2525: LibLine: Strange behavior when reaching end of bottom row
To reproduce: 
1) Get to the bottom of the terminal (press enter in the shell a bunch of times)
2) Type something until you get to the end of the line and try crossing into the next line

cc @alimpfard

### PR DESCRIPTION
**Pull request by alimpfard**
Fixes #2525

### CODE REVIEW

File: Libraries/LibLine/Editor.cpp [hunk 1]
```diff
@@ -992,6 +993,16 @@ void Editor::refresh_display()
         m_cached_buffer_size = m_buffer.size();
         return;
     }
+    // We might be at the last line, and have more than one line;
+    // Refreshing the display will cause the terminal to scroll,
+    // so note that fact and bring origin up.
+    auto current_num_lines = num_lines();
+    if (m_origin_x + current_num_lines > m_num_lines + 1) {
```
Comment (awesomekling) [inline Libraries/LibLine/Editor.cpp:1000]: Adding together x (horizontal) and lines (vertical) doesn't look right. What's happening here?

Comment (alimpfard) [reply Libraries/LibLine/Editor.cpp:1000]: the x and y are inverted here, x is the vertical axis, y is the horizontal axis.
kinda odd, but :shrug:

Comment (alimpfard) [reply Libraries/LibLine/Editor.cpp:1000]: I'm not even sure why I named them like this, I could go through and fix their names.
will take but a moment.

Comment (awesomekling) [reply Libraries/LibLine/Editor.cpp:1000]: Yeah let's fix that. This is way confusing :)

Comment (alimpfard) [reply Libraries/LibLine/Editor.cpp]: There we go.
no more confusion :)

Comment (alimpfard) [reply Libraries/LibLine/Editor.cpp]: ugh there's more of them T-T

**Pull request merged**